api/v1alpha1/clusterextension_types.go

```diff
@@ -106,6 +115,7 @@ func init() {
 		TypePackageDeprecated,
 		TypeChannelDeprecated,
 		TypeBundleDeprecated,
+		TypeProgressing,
```

Comment:
Hi @everettraven , actually it will fail another test and ask me to add it to the list. What I did was adding that to the list but then it requires me to actually set progressing status in the ClusterExtension controller. That part I'm honestly not sure if I did right. I appreciate some eyes on this. On the other hand if you really want to keep the type out of the list of conditions sets then we may have to change some tests. Happy to go either way. I count on your directions here.

Reply:
Oh that is interesting. I didn't realize there was a test that we had the ensured those constants are added to the conditionsets. 

I'm not sure this is really a valuable test for us at the moment. I'm not sure what the best path forward here is personally, maybe:
- Merge #842 
- Rebase this PR on ^, so it essentially becomes shifting the logic around but the clusterextension_controller already uses everything so we don't have to fight with CI

Reply:
Makes sense. I think we can talk about it on our meeting on Tuesday.

Reply:
> actually it will fail another test and ask me to add it to the list.

What was the specific test failure? I think @everettraven is correct that we should be able to let this status condition type be completely removed since it is currently only present in the Extension API.

Reply:
@joelanford I think it is these test cases: https://github.com/operator-framework/operator-controller/blob/e9acbd319c59e3805e5795a5f60a3fff9f24ccf9/api/v1alpha1/clusterextension_types_test.go#L18-L63